### PR TITLE
smart signal extension template

### PR DESCRIPTION
**Pull request by efvink**
Add smartSignal template with custom debugger

### CODE REVIEW

File: src/SDK/SmartSignals/SmartSignals.sln [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals.sln]: >File [](start = 33, length = 4)

why did you add another solution?

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/Manifest.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿{
+  "id": "DefaultId",
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/Manifest.json]: >DefaultId [](start = 9, length = 9)

MySignal

Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/Manifest.json]: >DefaultId [](start = 9, length = 9)

$safeprojectname$

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/Manifest.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  "name": "Default name",
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/Manifest.json]: >Default name [](start = 11, length = 12)

$safeprojectname$

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/Manifest.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  "description": "Default description",
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/Manifest.json]: >Default description [](start = 18, length = 19)

Signal description

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/Manifest.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+  "version": "1.0",
+  "assemblyName": "$safeprojectname$.dll",
+  "className": "$safeprojectname$.SmartSignal",
+  "supportedResourceTypes": [ "VirtualMachine" ],
+  "supportedCadencesInMinutes": [ 1440, 60 ]
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/Manifest.json]: >1440, 60 [](start = 34, length = 8)

put 5 here

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignal.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
+﻿using System.Linq;
+using Microsoft.Azure.Monitoring.SmartSignals;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace $safeprojectname$
+{
+	public class SmartSignal: ISmartSignal
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignal.cs]: >SmartSignal [](start = 14, length = 11)

add comment to the class

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignal.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    {   
+        public Task<SmartSignalResult> AnalyzeResourcesAsync(AnalysisRequest analysisRequest, ITracer tracer, CancellationToken cancellationToken)
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignal.cs]: >AnalyzeResourcesAsync [](start = 39, length = 21)

copy the comment from the interface

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignal.cs (hunk 6 above)
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignal.cs]: >: [](start = 25, length = 1)

space before : please

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignal.cs [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+        {
+            SmartSignalResult smartSignalResult = new SmartSignalResult();
+            smartSignalResult.ResultItems.Add(new SmartSignalResultItem("title", analysisRequest.TargetResources.First()));
+            return Task.FromResult(smartSignalResult);
+        }
+
+    }
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignal.cs]: remove extra line

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignalResultItem.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.Azure.Monitoring.SmartSignals;
+
+namespace $safeprojectname$
+{
+	public class SmartSignalResultItem : Microsoft.Azure.Monitoring.SmartSignals.SmartSignalResultItem
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignalResultItem.cs]: >SmartSignalResultItem [](start = 14, length = 21)

comment

Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignalResultItem.cs]: >Microsoft.Azure.Monitoring.SmartSignals [](start = 38, length = 39)

remove

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignalResultItem.cs [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+    {
+        public SmartSignalResultItem(string title, ResourceIdentifier resourceIdentifier) : base(title, resourceIdentifier)
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignalResultItem.cs]: >SmartSignalResultItem [](start = 15, length = 21)

add comment

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignals.csproj [hunk 11]
```diff
@@ -0,0 +1,27 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignals.csproj]: >Project [](start = 1, length = 7)

add stylecop and treat warnings as errors (copy from another project)

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignals.csproj [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+  <PropertyGroup Label="Globals">
+    <CustomProjectExtensionsPath Condition="Exists('$(LocalAppData)\CustomProjectSystems\SmartSignals\SmartSignals.props')">$(LocalAppData)\CustomProjectSystems\SmartSignals\</CustomProjectExtensionsPath>
+    <CustomProjectExtensionsPath Condition="'$(CustomProjectExtensionsPath)' == ''">$(MSBuildExtensionsPath)\CustomProjectSystems\SmartSignals\</CustomProjectExtensionsPath>
+  </PropertyGroup>
+  
+  <Import Project="$(CustomProjectExtensionsPath)SmartSignals.props" />
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignals.csproj]: >SmartSignals.props [](start = 49, length = 18)

how does it get there?

**Comment by EranGon** [comment]
Please ask Harel for a proper icon

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignals.vstemplate [hunk 13]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<VSTemplate Version="3.0.0" Type="Project" xmlns="http://schemas.microsoft.com/developer/vstemplate/2005" xmlns:sdk="http://schemas.microsoft.com/developer/vstemplate-sdkextension/2010">
+  <TemplateData>
+    <Name>SmartSignal Project</Name>
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignals.vstemplate]: >SmartSignal [](start = 10, length = 11)

Smart Signal

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignals.vstemplate [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+    <Description>A template for creating a SmartSignal project.</Description>
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignals.vstemplate]: >SmartSignal [](start = 43, length = 11)

Smart Signal

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignals.vstemplate [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+    <Icon>SmartSignals.ico</Icon>
+    <ProjectType>SmartSignals</ProjectType>
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignals.vstemplate]: SmartSignal

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignals.vstemplate [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+    <CreateInPlace>true</CreateInPlace>
+  </TemplateData>
+  <TemplateContent>
+    <Project File="SmartSignals.csproj" ReplaceParameters="true">
+      <!-- 
+      These are the items which will show up when a new project is created using the custom project system.
+      Items referenced in here must also be referenced in the project file.
+      -->
+      <ProjectItem ReplaceParameters="true" OpenInEditor="true">SmartSignal.cs</ProjectItem>
+      <ProjectItem ReplaceParameters="true" OpenInEditor="false">SmartSignalResultItem.cs</ProjectItem>
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignals.vstemplate]: >false [](start = 58, length = 5)

true

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/MyConfiguredProject.cs [hunk 17]
```diff
@@ -0,0 +1,34 @@
+﻿/***************************************************************************
+
+Copyright (c) Microsoft Corporation. All rights reserved.
+This code is licensed under the Visual Studio SDK license terms.
+THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
+ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
+IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
+PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
+
+***************************************************************************/
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/MyConfiguredProject.cs]: please update the copyright (copy from other projects)

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/MyUnconfiguredProject.cs [hunk 18]
```diff
@@ -0,0 +1,78 @@
+﻿/***************************************************************************
+
+Copyright (c) Microsoft Corporation. All rights reserved.
+This code is licensed under the Visual Studio SDK license terms.
+THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
+ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
+IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
+PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
+
+***************************************************************************/
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/MyUnconfiguredProject.cs]: update

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/MyConfiguredProject.cs [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+    using System.Threading.Tasks;
+    using System.Threading.Tasks.Dataflow;
+    using Microsoft.VisualStudio.ProjectSystem;
+    using Microsoft.VisualStudio.Shell;
+    using Microsoft.VisualStudio.Shell.Interop;
+    using Task = System.Threading.Tasks.Task;
+
+    [Export]
+    [AppliesTo(MyUnconfiguredProject.UniqueCapability)]
+    internal class MyConfiguredProject
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/MyConfiguredProject.cs]: >MyConfiguredProject [](start = 19, length = 19)

SmartSignalConfiguredProject

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/MyUnconfiguredProject.cs [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+    using Microsoft.VisualStudio.ProjectSystem.VS;
+    using Microsoft.VisualStudio.Shell;
+    using Microsoft.VisualStudio.Shell.Interop;
+    using Task = System.Threading.Tasks.Task;
+
+    [Export]
+    [AppliesTo(MyUnconfiguredProject.UniqueCapability)]
+    [ProjectTypeRegistration(VsPackage.ProjectTypeGuid, "SmartSignals", "#2", ProjectExtension, Language, resourcePackageGuid: VsPackage.PackageGuid, PossibleProjectExtensions = ProjectExtension, ProjectTemplatesDir = @"..\..\Templates\Projects\MyCustomProject")]
+    [ProvideProjectItem(VsPackage.ProjectTypeGuid, "My Items", @"..\..\Templates\ProjectItems\MyCustomProject", 500)]
+    internal class MyUnconfiguredProject
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/MyUnconfiguredProject.cs]: >MyUnconfiguredProject [](start = 19, length = 21)

SmartSignalUnconfiguredProject

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/SmartSignals.ProjectType.csproj [hunk 21]
```diff
@@ -0,0 +1,238 @@
...
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <SchemaVersion>2.0</SchemaVersion>
+    <ProjectTypeGuids>{82b43b9b-a64c-4715-b499-d71e9ca2bd60};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+    <ProjectGuid>{6D361361-3A0D-4168-B934-04A303C04636}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>SmartSignals</RootNamespace>
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/SmartSignals.ProjectType.csproj]: >SmartSignals [](start = 19, length = 12)

Microsoft.Azure.Monitoring.SmartSignals.ProjectType

please update all namespaces in this project

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/SmartSignals.ProjectType.csproj [hunk 22]
```diff
@@ -0,0 +1,238 @@
...
+    <AssemblyName>SmartSignals</AssemblyName>
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/SmartSignals.ProjectType.csproj]: >SmartSignals [](start = 18, length = 12)

does it have to match the project type?

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/source.extension.vsixmanifest [hunk 23]
```diff
@@ -0,0 +1,23 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<PackageManifest Version="2.0.0" xmlns="http://schemas.microsoft.com/developer/vsx-schema/2011" xmlns:d="http://schemas.microsoft.com/developer/vsx-schema-design/2011">
+  <Metadata>
+    <Identity Id="SmartSignals.ProjectType..01737f2f-e204-4e59-9d2e-b8d8f138f74c" Version="1.0" Language="en-US" Publisher="SmartSignals.ProjectType" />
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/source.extension.vsixmanifest]: >SmartSignals.ProjectType [](start = 124, length = 24)

Microsoft corp.

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/source.extension.vsixmanifest [hunk 24]
```diff
@@ -0,0 +1,23 @@
...
+    <DisplayName>SmartSignals VSIX</DisplayName>
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/source.extension.vsixmanifest]: >SmartSignals [](start = 17, length = 12)

Smart Signal

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/source.extension.vsixmanifest [hunk 25]
```diff
@@ -0,0 +1,23 @@
...
+    <Description xml:space="preserve">A template for creating a SmartSignal project.</Description>
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/source.extension.vsixmanifest]: >SmartSignal [](start = 64, length = 11)

Smart Signal

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/VsPackage.cs [hunk 26]
```diff
@@ -0,0 +1,54 @@
+﻿/***************************************************************************
+
+Copyright (c) Microsoft Corporation. All rights reserved.
+This code is licensed under the Visual Studio SDK license terms.
+THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
+ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
+IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
+PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
+
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/VsPackage.cs]: update

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/VsPackage.cs [hunk 27]
```diff
@@ -0,0 +1,54 @@
...
+        /// <summary>
+        /// The file extension of this project type.  No preceding period.
+        /// </summary>
+        public const string ProjectExtension = "csproj";
+
+        /// <summary>
+        /// The default namespace this project compiles with, so that manifest
+        /// resource names can be calculated for embedded resources.
+        /// </summary>
+        internal const string DefaultNamespace = "SmartSignals";
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/VsPackage.cs]: >DefaultNamespace [](start = 30, length = 16)

update

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/BuildSystem/Rules/SmartSignalsDebugger.xaml [hunk 28]
```diff
@@ -0,0 +1,24 @@
...
+	DisplayName="SmartSignals Emulator"
+	PageTemplate="debugger"
+	Description="SmartSignalsDebugger options"
+	xmlns="http://schemas.microsoft.com/build/2009/properties">
+    <Rule.DataSource>
+        <DataSource Persistence="UserFileWithXamlDefaults" HasConfigurationCondition="True"/>
+    </Rule.DataSource>
+
+    <StringProperty Name="SmartSignalsDebuggerCommand" DisplayName="Command"
+                Default="$(NuGetPackageRoot)microsoft.azure.monitoring.smartalerts\1.0.0-beta\tools\net461\AppEmulator\SmartSignalsRuntimeAppEmulator.exe"
```
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectType/BuildSystem/Rules/SmartSignalsDebugger.xaml]: >1.0.0-beta [](start = 83, length = 10)

that is a problem - what if the version changes?

File: src/SDK/SmartSignalsRuntimeAppEmulator/SmartSignalsRuntimeAppEmulator.csproj [hunk 29]
```diff
@@ -10,6 +10,7 @@
     <LanguageTargets>$(MSBuildToolsPath)\Microsoft.CSharp.targets</LanguageTargets>
   </PropertyGroup>
   <ItemGroup>
+    <PackageReference Include="NuProj.Common" Version="0.11.30" />
```
Comment (EranGon) [inline src/SDK/SmartSignalsRuntimeAppEmulator/SmartSignalsRuntimeAppEmulator.csproj]: why?

File: src/SDK/SmartSignalsSDK/SmartSignalsSDK.csproj [hunk 30]
```diff
@@ -11,7 +11,6 @@
     <PackageReference Include="NuProj" Version="0.11.30" />
     <PackageReference Include="NuProj.Common" Version="0.11.30" />
     <PackageReference Include="StyleCop.MSBuild" Version="5.0.0" />
-    <PackageReference Include="System.Net.Http" Version="4.3.3" />
```
Comment (EranGon) [inline src/SDK/SmartSignalsSDK/SmartSignalsSDK.csproj]: why?

Comment (efvink) [reply src/SDK/SmartSignalsSDK/SmartSignalsSDK.csproj]: because it's not in use

---
In reply to: [168188472](https://github.com/liadtal/Smart-Alerts/pull/75#discussion_r168188472) [](ancestors = 168188472)

File: src/SDK/SmartSignalsRuntimeAppEmulator/SmartSignalsRuntimeAppEmulator.csproj (hunk 29 above)
Comment (efvink) [reply src/SDK/SmartSignalsRuntimeAppEmulator/SmartSignalsRuntimeAppEmulator.csproj]: We need to install this package to all non-NuProj projects that are directly or indirectly referenced by the NuProj project ([https://github.com/nuproj/nuproj/blob/master/docs/Build.md](https://github.com/nuproj/nuproj/blob/master/docs/Build.md))

---
In reply to: [168188445](https://github.com/liadtal/Smart-Alerts/pull/75#discussion_r168188445) [](ancestors = 168188445)

File: src/SDK/SmartSignalsPackage/SmartSignalsPackage.nuproj [hunk 31]
```diff
@@ -37,14 +37,18 @@
...
     <File Include="..\SmartSignalsBuildTask\build\**.*">
       <TargetPath>build\</TargetPath>
     </File>
+    <File Include="..\SmartSignalsBuildTask\$(Outdir)\net461\**.*">
+      <TargetPath>tools\net461\BuildTask</TargetPath>
+    </File>
+    <ProjectReference Include="..\SmartSignalsSDK\SmartSignalsSDK.csproj">
+      <PackageDirectory>lib</PackageDirectory>
+    </ProjectReference>
+    <File Include="..\SmartSignalsRuntimeAppEmulator\$(Outdir)\net461\**.*">
```
Comment (aviled) [inline src/SDK/SmartSignalsPackage/SmartSignalsPackage.nuproj]: >SmartSignalsRuntimeAppEmulator [](start = 22, length = 30)

Why do we need the emulator build output in the nuproj?

File: src/SDK/SmartSignalsRuntimeAppEmulator/SmartSignalsRuntimeAppEmulator.csproj (hunk 29 above)
Comment (aviled) [reply src/SDK/SmartSignalsRuntimeAppEmulator/SmartSignalsRuntimeAppEmulator.csproj]: but why is the nuproj project reference the emulator? seems like it should be the other way around, no?

---
In reply to: [168402832](https://github.com/liadtal/Smart-Alerts/pull/75#discussion_r168402832) [](ancestors = 168402832,168188445)

File: src/SDK/SmartSignalsPackage/SmartSignalsPackage.nuproj (hunk 31 above)
Comment (efvink) [reply src/SDK/SmartSignalsPackage/SmartSignalsPackage.nuproj]: The template uses the package, and the emulator is set as the debugger. 

---
In reply to: [168441615](https://github.com/liadtal/Smart-Alerts/pull/75#discussion_r168441615) [](ancestors = 168441615)

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignalResultItem.cs (hunk 9 above)
Comment (efvink) [reply src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignalResultItem.cs]: Without the prefix the derived class has the same name as the base class. 

---
In reply to: [168183404](https://github.com/liadtal/Smart-Alerts/pull/75#discussion_r168183404) [](ancestors = 168183404)

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignals.csproj (hunk 11 above)
Comment (efvink) [reply src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignals.csproj]: as we discussed, I won't add stylecop , but I add some style

---
In reply to: [168183589](https://github.com/liadtal/Smart-Alerts/pull/75#discussion_r168183589) [](ancestors = 168183589)

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignalResultItem.cs (hunk 9 above)
Comment (EranGon) [reply src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignalResultItem.cs]: MySmartSignalResultItem

---
In reply to: [168483832](https://github.com/liadtal/Smart-Alerts/pull/75#discussion_r168483832) [](ancestors = 168483832,168183404)

File: src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignal.cs (hunk 6 above)
Comment (EranGon) [inline src/SDK/SmartSignals/SmartSignals/SmartSignals.ProjectTemplate/SmartSignal.cs]: >SmartSignal [](start = 14, length = 11)

MySmartSignal

File: src/SDK/SmartSignalsPackage/build/net461/Microsoft.Azure.Monitoring.SmartAlerts.targets [hunk 32]
```diff
@@ -1,8 +1,9 @@
 ﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <_SmartAlertsTaskFramework Condition=" '$(_SmartAlertsTaskFramework)' == ''">net461</_SmartAlertsTaskFramework>
-    <_SmartAlertsTasksDir Condition=" '$(_SmartAlertsTasksDir)'=='' ">$(MSBuildThisFileDirectory)..\..\tools\$(_SmartAlertsTaskFramework)\</_SmartAlertsTasksDir>
+    <_SmartAlertsFramework Condition=" '$(_SmartAlertsTaskFramework)' == ''">net461</_SmartAlertsFramework>
+    <_SmartAlertsTasksDir Condition=" '$(_SmartAlertsTasksDir)'=='' ">$(MSBuildThisFileDirectory)..\..\tools\$(_SmartAlertsFramework)\BuildTask\</_SmartAlertsTasksDir>
```
Comment (EranGon) [inline src/SDK/SmartSignalsPackage/build/net461/Microsoft.Azure.Monitoring.SmartAlerts.targets]: >..\.. [](start = 97, length = 5)

this is not right
